camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/reporter/StatsdReporter.java

```diff
@@ -40,15 +45,28 @@ private void submitCountersToStatsd(Job job) throws IOException {
         }
     }
 
-    public static Boolean getStatsdEnabled(Job job) {
-        return job.getConfiguration().getBoolean(STATSD_ENABLED, false);
+    public static void logEventDelay(JobContext context, String topic, CamusWrapper wrap) {
+        if (getStatsdEnabled(context.getConfiguration())) {
+            StatsDClient statsd = new NonBlockingStatsDClient(
```

Comment:
You are instantiating `statsd` for every single record it sees and then you only use it if the delay is longer than a hour.  Why not instantiate `statsd` once inside `EtlRecordReader`? That might mean a small refactor but I think this is very expensive. If the refactor is not feasible then  at the very least, you should only instantiate `statsd` if the delay is more than a hour.